tsconfig.base.esm.json

```diff
@@ -2,6 +2,6 @@
   "extends": "./tsconfig.base.es5.json",
   "compilerOptions": {
     "module": "ES6",
-    "moduleResolution": "node"
+    "moduleResolution": "node10"
```

Comment:
We specifically need `node10` and not `node16`?
https://www.typescriptlang.org/tsconfig/#moduleResolution
(I am pretty ignorant here, but it seems odd given that our base supported Node.js version for the "next" branch will be node 18.)

Reply:
I thought once we have TS updated we could do this change within the scope of https://github.com/open-telemetry/opentelemetry-js/issues/4898 but I'm okay to change it

Reply:
at the end it was necessary even for `tsconfig.base.json` file. So now they all have `node16`